p2p/libp2p/discovery/hostDecorator.go

```diff
@@ -0,0 +1,91 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+const (
+	hardCPSLimitLow  = 2
+	hardCPSLimitHigh = 100
+)
+
+type hostDecorator struct {
+	host.Host
+	acceptedChan chan struct{}
+	timeout      time.Duration
+}
+
+// NewHostDecorator creates a new decorator around an existing host (h) that will only allow
+// cps connections per second and drop the connection request upon timeout expiration
+func NewHostDecorator(h host.Host, ctx context.Context, cps uint32, timeout time.Duration) (*hostDecorator, error) {
+	if h == nil {
+		return nil, p2p.ErrNilHost
+	}
+
+	if ctx == nil {
+		return nil, p2p.ErrNilContext
+	}
+
+	if timeout < 0 {
+		return nil, p2p.ErrInvalidDurationProvided
+	}
+
+	if cps < hardCPSLimitLow {
+		cps = hardCPSLimitLow
+	}
+
+	if cps > hardCPSLimitHigh {
+		cps = hardCPSLimitHigh
```

Comment:
Same as above here. We should return an (nil, error) if the cps is not in the hardCPSLimitHigh and hardCPSLimitLow.

Reply:
:) the convention "was" or "is" ? 
if the rational is to stop the exacution for invalid configs "panic" should be ta way to go, that way you also get a location report (position in code and execution (routine etc)).

Changed, but still ...

Reply:
the convention is :)